eos-tech-support/eos-stage-ostree

```diff
@@ -0,0 +1,120 @@
+    # Check for an update every time the updater runs
+    interval_days=0
+elif [ "$STAGE" == "demo" ] ; then
+    base_url="https://ostree.endlessm.com/staging/demo"
+    # Get the OS product version (e.g., eos3)
+    version=$(. /etc/os-release && echo $VERSION | cut -d. -f1)
+    # Check for an update every time the updater runs
+    interval_days=0
+elif [ "$STAGE" == "prod" ] ; then
+    base_url="https://ostree.endlessm.com/ostree"
+    # Get the OS product version (e.g., eos3)
+    version=$(. /etc/os-release && echo $VERSION | cut -d. -f1)
+    # By default we check for updates every two weeks on production
+    interval_days=14
+else
+    echo "Invalid STAGE -- must be dev, demo, or prod" >&2
+    exit 1
+fi
+
+# Get the current refspec.
+refspec=$(ostree admin status | awk '/refspec:/{print $3}' | head -n1)
+branch=${refspec#*:}
+
+# Construct the new branch with the product or product.major version
+# depending on the stage.
+new_branch="${branch%/*}/eos${version}"
+
+# Get the current URL and convert to the new stage.
+url=$(ostree config get 'remote "eos".url')
+repo=${url##*/}
+new_url="${base_url}/${repo}"
+
+# Make sure the autoupdater doesn't start the updater after it's
+# killed below.
+echo "Stopping running eos-autoupdater."
+systemctl stop eos-autoupdater.timer eos-autoupdater.service
+
+echo "Killing eos-updater."
+systemctl stop eos-updater.service
+
+echo "Configuring OSTree for $STAGE stage."
+# HACK! HACK! HACK! ostree admin set-origin is horribly broken
+# as of ostree 2015.7. It won't change the url or use any --set
+# options for an existing remote. Instead, use the config
+# builtin to handle that. The correct set-origin command is left
+# in place in case this is ever fixed. Upstream bug at
+# https://bugzilla.gnome.org/show_bug.cgi?id=753373
+ostree config set 'remote "eos".url' "$new_url"
+ostree config set 'remote "eos".branches' "${new_branch};"
+ostree admin set-origin eos "$new_url" "$new_branch" \
+       --set=branches="${new_branch};"
+
+# Adjust the frequency of update checks
+# eos-updater.conf is the old location of the config file.
+sed -i "s/IntervalDays=.*/IntervalDays=$interval_days/" /etc/eos-updater/eos-autoupdater.conf 2>/dev/null || true
+sed -i "s/IntervalDays=.*/IntervalDays=$interval_days/" /etc/eos-updater.conf 2>/dev/null || true
+
+if [ ! -f /etc/eos-updater/eos-autoupdater.conf ] && [ ! -f /etc/eos-updater.conf ]; then
+    mkdir -p /etc/eos-updater
+    cp /usr/share/eos-updater/eos-autoupdater.conf /etc/eos-updater
+
+    sed -i "s/IntervalDays=.*/IntervalDays=$interval_days/" /etc/eos-updater/eos-autoupdater.conf
+fi
+
+# Restart the autoupdater timer, and try to update now.
+# Note that downgrades (e.g., from demo to previous prod release)
+# won't happen automatically.  They would have to be done via
+# `ostree admin upgrade --allow-downgrade`, but that should be
+# a manual step if someone really wants to downgrade.
```

Comment:
Note that the "downgrade" may not really be a downgrade if you're switching stages since "downgrade" just means the commit is older than what you have. If you switch from dev to demo and the demo commit happened to have been made before the dev commit (likely since we rarely promote dev to demo), then there will be no update.